bootstrap/cmd/bootstrap/app/gcpUtils.go

```diff
@@ -308,3 +310,55 @@ func (s *ksServer) ApplyIamPolicy(ctx context.Context, req ApplyIamRequest) erro
 	}, exp)
 	return nil
 }
+
+// ProjectAccessChecker defines a func that can be used to check project access
+type ProjectAccessChecker func(string, oauth2.TokenSource) (bool, error)
+
+// CheckProjectAccess verifies whether the supplied token provides access to the
+// indicated project. A false could indicate the credential provides insufficient
+// privileges or is expired
+func CheckProjectAccess(project string, ts oauth2.TokenSource) (bool, error) {
+	ctx := context.Background()
+
+	s, err := crm.NewService(ctx, option.WithTokenSource(ts))
+
+	if err != nil {
+		log.Errorf("Failed to create service with error; %+v", err)
+		return false, err
+	}
+
+	p := crm.NewProjectsService(s)
+
+	// TODO(jlewi): We use setIamPolicy as a check that we have sufficient access to the project
+	// might be better to use cluster Admin or similar permission.
+	req := &crm.TestIamPermissionsRequest{
+		Permissions: []string{"resourcemanager.projects.setIamPolicy"},
+	}
+
+	exp := backoff.NewExponentialBackOff()
+	exp.InitialInterval = 2 * time.Second
+	exp.MaxInterval = 5 * time.Second
+	exp.MaxElapsedTime = time.Minute
+	exp.Reset()
```

Comment:
Should we decrease retry numbers considering cloudresourcemanager has API quota limit? (100 per project per minute?)